Signal/src/ViewControllers/ConversationView/Cells/OWSMessageBubbleView.h

```diff
@@ -0,0 +1,46 @@
+//
+//  Copyright (c) 2018 Open Whisper Systems. All rights reserved.
+//
+
+NS_ASSUME_NONNULL_BEGIN
+
+@class ConversationViewItem;
+
+typedef NS_ENUM(NSUInteger, OWSMessageGestureLocation) {
+    // Message text, etc.
+    OWSMessageGestureLocation_Default,
+    OWSMessageGestureLocation_OversizeText,
+    OWSMessageGestureLocation_Media,
+    OWSMessageGestureLocation_QuotedReply,
+};
+
+@interface OWSMessageBubbleView : UIView
+
+@property (nonatomic, nullable) ConversationViewItem *viewItem;
+
+@property (nonatomic) int contentWidth;
+
+@property (nonatomic) NSCache *cellMediaCache;
+
+@property (nonatomic, nullable, readonly) UIView *lastBodyMediaView;
+
+- (instancetype)init NS_UNAVAILABLE;
+
+- (instancetype)initWithFrame:(CGRect)frame NS_DESIGNATED_INITIALIZER;
+
+- (instancetype)initWithCoder:(NSCoder *)coder NS_UNAVAILABLE;
+
+- (void)configureViews;
```

Comment:
This view can be used for rendering or measurement. This method only applies to rendering.